hopr/hopr-lib/src/lib.rs

```diff
@@ -177,26 +178,44 @@ impl HoprLibProcesses {
     }
 }
 
+/// Creates a pipeline that chains the indexer generated data, processes them into
+/// the individual components and creates a filtered output stream that is fed into
+/// the transport layer swarm.
+///
+/// * `event_stream` - represents the events generated by the indexer. If indexer is not synced,
+/// it will not generate any events.
+/// * `preloading_event_stream` - a stream used by the components to preload the data from the objects
+/// (db, channel graph...)
 #[allow(clippy::too_many_arguments)] // TODO: refactor this function into a reasonable group of components once fully rearchitected
-pub async fn chain_event_refresh_process<Db, S, T>(
+pub async fn chain_events_to_transport_events<StreamIn, StreamInPreloading, Db>(
+    event_stream: StreamIn,
+    preloading_event_stream: StreamInPreloading,
     me: PeerId,
     me_onchain: Address,
     db: Db,
     multi_strategy: Arc<MultiStrategy>,
-    event_stream: S,
     channel_graph: Arc<RwLock<core_path::channel_graph::ChannelGraph>>,
-    transport_indexer_actions: core_transport::IndexerActions,
     indexer_action_tracker: Arc<IndexerActionTracker>,
-    network: Arc<Network<T>>,
-) where
-    Db: HoprDbAllOperations + Send + Sync + 'static,
-    S: Stream<Item = SignificantChainEvent> + Send + 'static,
-    T: HoprDbAllOperations + Sync + Send + std::fmt::Debug + 'static,
+    network: Arc<Network<Db>>,
+) -> impl Stream<Item = PeerTransportEvent> + Send + 'static
+where
+    Db: HoprDbAllOperations + Clone + Send + Sync + std::fmt::Debug + 'static,
+    StreamIn: Stream<Item = SignificantChainEvent> + Send + 'static,
+    StreamInPreloading: Stream<Item = IndexerTransportEvent> + Send + 'static,
 {
-    pin_mut!(event_stream);
-    while let Some(event) = event_stream.next().await {
+    let network_clone = network.clone();
+
+    Box::pin(event_stream.filter_map(move |event| {
+        let db = db.clone();
+        let multi_strategy = multi_strategy.clone();
+        let channel_graph = channel_graph.clone();
+        let indexer_action_tracker = indexer_action_tracker.clone();
+        let network = network.clone();
```

Comment:
Why not use the `network_clone` from the top or at least clone from it like you did later in the function?

Reply:
Because 1 clone per separate closure is needed.